cli/src/main/kotlin/commands/UploadResultToSw360Command.kt

```diff
@@ -0,0 +1,189 @@
+import org.eclipse.sw360.antenna.http.config.HttpClientConfig
+import org.eclipse.sw360.antenna.sw360.client.adapter.SW360ConnectionFactory
+import org.eclipse.sw360.antenna.sw360.client.adapter.SW360ProjectClientAdapter
+import org.eclipse.sw360.antenna.sw360.client.adapter.SW360ReleaseClientAdapter
+import org.eclipse.sw360.antenna.sw360.client.config.SW360ClientConfig
+import org.eclipse.sw360.antenna.sw360.client.rest.resource.SW360Visibility
+import org.eclipse.sw360.antenna.sw360.client.rest.resource.projects.SW360Project
+import org.eclipse.sw360.antenna.sw360.client.rest.resource.releases.SW360Release
+import org.eclipse.sw360.antenna.sw360.client.utils.SW360ClientException
+
+import org.ossreviewtoolkit.GlobalOptions
+import org.ossreviewtoolkit.model.OrtResult
+import org.ossreviewtoolkit.model.Package
+import org.ossreviewtoolkit.model.Project
+import org.ossreviewtoolkit.model.config.Sw360StorageConfiguration
+import org.ossreviewtoolkit.model.jsonMapper
+import org.ossreviewtoolkit.model.readValue
+import org.ossreviewtoolkit.utils.collectMessagesAsString
+import org.ossreviewtoolkit.utils.expandTilde
+import org.ossreviewtoolkit.utils.formatSizeInMib
+import org.ossreviewtoolkit.utils.log
+import org.ossreviewtoolkit.utils.perf
+
+class UploadResultToSw360Command : CliktCommand(
+    name = "upload-result-to-sw360",
+    help = "Upload an ORT result to SW360.",
+    epilog = "EXPERIMENTAL: The command is still in development and usage will likely change in the near future. The " +
+            "command expects that a Sw360Storage for the scanner is configured."
+) {
+    private val ortFile by option(
+        "--ort-file", "-i",
+        help = "The ORT result file to read as input."
+    ).convert { it.expandTilde() }
+        .file(mustExist = true, canBeFile = true, canBeDir = false, mustBeWritable = false, mustBeReadable = true)
+        .convert { it.absoluteFile.normalize() }
+        .required()
+        .inputGroup()
+
+    private val sw360JsonMapper = jsonMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
+    private val globalOptionsForSubcommands by requireObject<GlobalOptions>()
+
+    override fun run() {
+        val (ortResult, duration) = measureTimedValue { ortFile.readValue<OrtResult>() }
+
+        log.perf {
+            "Read ORT result from '${ortFile.name}' (${ortFile.formatSizeInMib}) in ${duration.inMilliseconds}ms."
+        }
+
+        val sw360Config = globalOptionsForSubcommands.config.scanner?.storages?.values
+            ?.filterIsInstance<Sw360StorageConfiguration>()?.singleOrNull()
+
+        requireNotNull(sw360Config) {
+            "No SW360 storage is configured for the scanner."
+        }
+
+        val httpClientConfig = HttpClientConfig
+            .basicConfig()
+            .withObjectMapper(sw360JsonMapper)
+        val newHttpClient = HttpClientFactoryImpl().newHttpClient(httpClientConfig)
+
+        requireNotNull(newHttpClient) {
```

Comment:
The factory will never return a *null* object. Is this check done to deal with the Java types (!)? Is there an easier way?